internal/test/ledger/ledger.go

```diff
@@ -51,7 +51,11 @@ func (ls MockLedgerState) StakeRegistration(
 ) ([]common.StakeRegistrationCertificate, error) {
 	ret := []common.StakeRegistrationCertificate{}
 	for _, cert := range ls.MockStakeRegistration {
-		if string(cert.StakeRegistration.Credential) == string(stakingKey) {
+		if string(
+			common.Blake2b224(cert.StakeRegistration.Credential).Bytes(),
```

Comment:
You can probably avoid this whole dance with `[]byte(cert.StakeRegistration.Credential)`

Reply:
I kept this because it's what's being done in the other test.